stock_dynamic_routing/models/stock_move.py

```diff
@@ -273,13 +274,56 @@ def _apply_routing_rule_pull(self, routing_details):
 
             pickings_to_check_for_emptiness |= move.picking_id
             move._assign_picking()
-            # Note: we have to call _action_assign() here because if the move
-            # has been split because of partial availability, we want to ensure
-            # to reserve the move which has been "routed" first. Even if
-            # _action_assign() is called again, it should not be an issue
-            # because the move's state will be "assigned" and will be excluded
-            # by the method.
-            move.with_context(exclude_apply_dynamic_routing=True)._action_assign()
+            move_ids_to_assign_per_location[move.location_id].append(move.id)
+
+        # We have two kind of "routed" moves:
+        #
+        # * The ones where only the picking type / destination has been changed
+        # * The ones which are assigned to a new picking and a new move is inserted
+        #   to reach the destination
+        #
+        # For these 2 kinds, we have to call _action_assign() here because if
+        # the move has been split because of partial availability, we want to
+        # ensure to reserve the move which has been "routed" before the moves
+        # without routing. Even if _action_assign() is called again, it will
+        # not be an issue (unless a custom module does not check the current
+        # state) because the move's state will be "assigned" and will be
+        # excluded by the method.
+        #
+        # Now among these moves, some may have their source location changed,
+        # for instance:
+        #
+        # * A move of 20 units has a source location: Stock
+        # * It is split to take
+        #   * 10 in Stock/Shelving
+        #   * 10 in Stock/Reserve
+        # * Stock/Reserve insert a new move to do Stock/Reserve -> Stock/Shelving
```

Comment:
I'm not sure to understand this sentence, the `Stock/Reserve -> Stock/Shelving` part especially?

Reply:
I'll improve the example

Reply:
updated